teamcity: failed tests on master: roachtest/acceptance

The following tests appear to have failed:

[#912814](https://teamcity.cockroachdb.com/viewLog.html?buildId=912814):


```
--- FAIL: roachtest/acceptance (1040.040s)

--- FAIL: roachtest/acceptance/bank/cluster-recovery (600.025s)
	test.go:498,cluster.go:830,bank.go:224: /go/bin/roachprod start local:2 returned:
		stderr:
		write any data in your cluster.
		* - There is no network encryption nor authentication, and thus no confidentiality.
		* 
		* Check out how to secure your cluster: https://www.cockroachlabs.com/docs/v2.2/secure-a-cluster.html
		*
		*
		* ERROR: could not cleanup temporary directories from record file: could not lock temporary directory /home/roach/local/2/data/cockroach-temp142723177, may still be in use: IO error: While lock file: /home/roach/local/2/data/cockroach-temp142723177/TEMP_DIR.LOCK: Resource temporarily unavailable
		*
		Failed running "start"
		E180919 03:24:03.287349 1 cli/error.go:230  exit status 1
		Error: exit status 1
		Failed running "start"
		
		github.com/cockroachdb/roachprod/install.Cockroach.Start.func6
			/go/src/github.com/cockroachdb/roachprod/install/cockroach.go:371
		github.com/cockroachdb/roachprod/install.(*SyncedCluster).Parallel.func1.1
			/go/src/github.com/cockroachdb/roachprod/install/cluster_synced.go:1073
		runtime.goexit
			/usr/local/go/src/runtime/asm_amd64.s:2361: 
		2018/09/19 03:24:03 command failed
		
		stdout:
		local: starting
		: exit status 1
	test.go:779: test timed out (10m0s)
	test.go:498,bank.go:276,bank.go:357,acceptance.go:59: context canceled

--- FAIL: roachtest/acceptance (1040.040s)

--- FAIL: roachtest/acceptance/bank/cluster-recovery (600.025s)
	test.go:498,cluster.go:830,bank.go:224: /go/bin/roachprod start local:2 returned:
		stderr:
		write any data in your cluster.
		* - There is no network encryption nor authentication, and thus no confidentiality.
		* 
		* Check out how to secure your cluster: https://www.cockroachlabs.com/docs/v2.2/secure-a-cluster.html
		*
		*
		* ERROR: could not cleanup temporary directories from record file: could not lock temporary directory /home/roach/local/2/data/cockroach-temp142723177, may still be in use: IO error: While lock file: /home/roach/local/2/data/cockroach-temp142723177/TEMP_DIR.LOCK: Resource temporarily unavailable
		*
		Failed running "start"
		E180919 03:24:03.287349 1 cli/error.go:230  exit status 1
		Error: exit status 1
		Failed running "start"
		
		github.com/cockroachdb/roachprod/install.Cockroach.Start.func6
			/go/src/github.com/cockroachdb/roachprod/install/cockroach.go:371
		github.com/cockroachdb/roachprod/install.(*SyncedCluster).Parallel.func1.1
			/go/src/github.com/cockroachdb/roachprod/install/cluster_synced.go:1073
		runtime.goexit
			/usr/local/go/src/runtime/asm_amd64.s:2361: 
		2018/09/19 03:24:03 command failed
		
		stdout:
		local: starting
		: exit status 1
	test.go:779: test timed out (10m0s)
	test.go:498,bank.go:276,bank.go:357,acceptance.go:59: context canceled


```

Please assign, take a look and update the issue accordingly.

--- a/pkg/cmd/roachtest/test.go
+++ b/pkg/cmd/roachtest/test.go
@@ -776,12 +776,16 @@ func (r *registry) run(
 
 					select {
 					case <-time.After(timeout):
-						t.printf("test timed out (%s)", timeout)
+						t.printf("test timed out (%s)\n", timeout)
 						if c != nil {
 							c.FetchLogs(ctx)
 							// NB: c.destroyed is nil for cloned clusters (i.e. in subtests).
 							if !debugEnabled && c.destroyed != nil {
 								c.Destroy(ctx)
+							}
+							if local {
+								t.printf("waiting for test to tear down since cluster is local\n")
+								<-done
 							}
 						}
 					case <-done:

